File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        maxFreq = 0
        repeated = 0
        freqs = {}
        for num in nums:
            if num not in freqs:
                freqs[num] = 0
            freqs[num] += 1
            if freqs[num] > maxFreq:
                maxFreq = freqs[num]
                repeated = 1
            elif freqs[num] == maxFreq:
                repeated += 1
        return maxFreq * repeated

File: 3242-count-elements-with-maximum-frequency/3242-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        int maxFreq = 0;
        int repeated = 0;
        unordered_map<int, int> freqs;
        for(int i = 0; i < nums.size(); i++)
        {
            freqs[nums[i]]++;
            if(freqs[nums[i]] > maxFreq)
            {
                maxFreq = freqs[nums[i]];
                repeated = 1;
            }
            else if(freqs[nums[i]] == maxFreq)
            {
                repeated++;
            }
        }
        return maxFreq * repeated;
    }
};